### PR TITLE
Upgrade several dev dependencies

### PR DESCRIPTION
**Pull request by realityking**
#### :rocket: Why this change?

Update dev dependencies that are relatively safe. Hopefully this helps #985 getting closer to being merged.

Since these are all dev dependencies I put them down as `chore`. They don't need to trigger a release.

#### :white_check_mark: What didn't I forget?

- [ ] To write docs
- [ ] To write tests
- [X] To put [Conventional Changelog](https://dredd.readthedocs.io/en/latest/contributing/#sem-rel) prefixes in front of all my commits and run `npm run lint`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,25 +71,25 @@
     "babel-cli": "6.26.0",
     "babel-core": "6.26.0",
     "babel-preset-env": "1.6.1",
-    "body-parser": "1.17.1",
+    "body-parser": "1.18.2",
     "conventional-changelog-lint": "1.1.9",
-    "coveralls": "2.13.0",
+    "coveralls": "3.0.0",
```
Comment (honzajavorek) [inline package.json]: I'll check all upgrades from the licensing standpoint, but under this one I can comment right away.

`coveralls` do not have a license distributed with the package, nor they have the license in the GitHub repo. I tried to contribute a solution - https://github.com/nickmerwin/node-coveralls/pull/175, but there's no activity. When trying to figure out the copyright notice, I was surprised to find out the company uses two different names: "Coveralls, LLC" (on their site) and "LEMUR Heavy Industries" ([lemurheavy](https://github.com/lemurheavy) - author organization of the GitHub integration apps). There is no contact page on the coveralls.io website, no way to figure out who actually runs the business. Maybe it's just a lack of organization, maybe sloppiness, but it doesn't really build trust.

At the same time, the library itself is at @nickmerwin's personal GitHub profile, and the `author` field in `package.json` mentions someone called "Gregg Caines". The `maintainers` field mentions `nick@coveralls.io`, so I suppose it's not Nick's "hobby" project. Anyway, it's a mess. And I've seen a lot of npm packages, but this is one of the messiest I've seen.

I've tried to contact the maintainers over the GitHub issues and by email. No response. At about the same time (not implying anything though) the integration stopped to work under Dredd's Pull Requests. I tried to debug the issue like 10 times with no success. I tried to contact coveralls.io, no response.

Honestly, I think I'm giving up on Coveralls. While I believe the test coverage is very important, I also believe there are Coveralls competitors and we should go with them. So if I may, I'd kindly ask you not to upgrade the `coveralls` dependency and if you have a spare time, a Pull Request removing coveralls from the toolchain would be awesome.

I discovered following alternatives:

- https://www.codacy.com/
- https://codeclimate.com/
- https://codecov.io/

Do you have any experience with any of them?

Comment (realityking) [reply package.json]: Ouch, that’s some deep looking you did I to that. I’ll revert that part of the PR.

I can recommend codecov.io, it’s the solution we use at Contentful and IMHO it’s much better hab coveralls.

Comment (honzajavorek) [reply package.json]: Yeah, I really tried hard to fix it and to continue using Coveralls before I gave up 😄

Comment (realityking) [reply package.json]: Removed the coveralls update. I'd prefer not to do the removal PR, IMHO that should happen in one go with the set up of a new solution which I can't do.

**Comment by realityking** [comment]
@honzajavorek Would it help to split this into individual PRs?

**Comment by honzajavorek** [comment]
@realityking I don't think so. There are various license issues, which need to be resolved etc. I'm working on speeding up accepting pull requests like this, internally, and generally improving the situation so this is at least transparent and automated. Thanks for the patience.

**Comment by honzajavorek** [comment]
I created a new label called "licensing" with individual issues featuring details to each of the dependencies. While I'm very grateful for your work, I'm going to close this now and let's sort it out one by one, as you suggested originally. I'll upgrade all dependencies without issues myself as soon as possible.

**Comment by realityking** [comment]
Makes sense 👍

I'll give you a bit to sort things out with the easy stuff, then I'll jump back in 🙂

**Pull request closed without merging**